### DISCUSSION TITLE
Question about Wasm & clipboard

### DISCUSSION THREAD

**Discussion by f-saez** [emilk/egui#2171] [General]
Hi,

I've run into troubles with clipboard ans WASM.
For what I've seen, it works perfectly well with the demo hosted in github (FontBook), but I'm unable to make it works.
If I build and run egui_demo on my machine, it works. As soon as I try to use egui_demo from another machine on my local network, the clipboard stop working. Basically, I've just copied the content of docs directory, nothing more.
I've tried with apache or simple_http_server, it doesn't matter.
I must have forgotten or misunderstood something obvious.

If someone could give me a hint ...

**Comment by paul-hansen** [comment]
I think this is because the newer Web "Clipboard API" is only available when you are using HTTPS.

Source:
https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
![image](https://user-images.githubusercontent.com/7019130/197148047-04fa0511-56a5-468f-897d-70c6970beccc.png)

**Reply by f-saez** [reply]
It makes sense, I should have think of that.

thanks for your answer and for your time.

**Reply by xNxExOx** [reply]
I have same issue Ctrl+C (copy), and Ctrl+X (cut) does not work when compiled to web assembly.

For debug run `trunk serve` it the provided explanation would be good enough. _(is there easy way to enable it for debug?)_

But for release `trunk build --release --public-url wa/deck_info/` I use HTTPS https://jakub.bandola.cz/wa/deck_info/ So what else can be the issue?

**Reply by emilk** [reply]
You need `RUSTFLAGS="--cfg=web_sys_unstable_apis" trunk serve`. I've updated the instructions in https://github.com/emilk/eframe_template/

**Comment by coderedart** [comment]
As a temporary workaround,
create `.cargo/config.toml` file with the following contents
```toml
[build]
rustflags = ["--cfg=web_sys_unstable_apis"]
```

which should add the flag automatically
